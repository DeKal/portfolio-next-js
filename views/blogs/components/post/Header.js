import styled from 'styled-components/macro'

export default styled.h1`
  line-height: 48px;
  font-size: ${props => props.theme.header.fontSize};
  color: ${props => props.theme.colors.dark};
  font-weight: 400;
`
