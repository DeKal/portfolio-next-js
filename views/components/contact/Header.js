import React from 'react'
import styled from 'styled-components/macro'

const Header = () => (
  <Container className="contact-card__header">
    <h2>Contact information</h2>
    <h4>My pleasure to meet you!!</h4>
  </Container>
)

export default Header

const Container = styled.header`
  padding: 20px;
  background-color: ${props => props.theme.colors.colorAccent};
  color: ${props => props.theme.colors.colorLight};
`
