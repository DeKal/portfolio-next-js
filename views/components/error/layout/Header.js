import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

const Header = ({ children }) => {
  return (
    <Container>
      <Style>{children}</Style>
    </Container>
  )
}

Header.propTypes = {
  children: PropTypes.string
}

export default Header

const Container = styled.div`
  height: 280px;
  position: relative;
  z-index: -1;

  @media only screen and (max-width: 767px) {
    height: 142px;
  }
`
const Style = styled.h1`
  @media only screen and (max-width: 767px) {
    font-size: 112px;
  }

  font-size: 230px;
  margin: 0;
  font-weight: 900;
  position: absolute;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background: url(/images/oops.jpg) no-repeat;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-size: cover;
  background-position: center;
`
