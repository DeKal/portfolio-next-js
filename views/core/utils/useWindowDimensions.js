import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
