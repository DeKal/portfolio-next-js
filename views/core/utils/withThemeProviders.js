import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
