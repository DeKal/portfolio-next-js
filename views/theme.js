const theme = {
  font:
    '-apple-system, blinkmacsystemfont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
  navHeaderHeight: '40px',
  navContentMinHeight: '80px',
  colors: {
    // Color variables
    colorLight: '#ecf0f1',
    colorDarken: '#bdc3c7',
    colorDark: '#2c3e50',
    colorAccent: '#FFA726',
    // primary colors
    lightPrimary: '#fff',
    darkPrimary: '#000',

    // additional colors
    lightSecondary: '#f4f4f4',
    darkSecondary: '#909090',
    medium: '#dfdfdf',
    darkTertiary: '#6d6e6e',

    // feedback/highlight colors
    positiveDark: '#00b09f',
    positiveLight: '#d4ece7',
    neutral: '#9dabae',
    negativeDark: '#ba4242',
    negativeLight: '#ff8484',

    facebook: '#3a5795',
    facebookHover: '#2B4273',

    darkDisabled: '#6d6e6e',
    lightDisabled: '#cdece9',

    placeholder: '#f1f1f1',
    outlet: '#ab0030',

    // Colors from new styleguide
    grey100: '#262626',
    grey80: '#474747',
    grey60: '#737373',
    grey20: '#dedede',
    grey10: '#f4f4f4',

    green10: '#d9f7d2',
    green60: '#2e8618',
    green80: '#1a6307',
    green90: '#1b510e',

    red10: '#fde4e4',
    red60: '#b81818',
    red80: '#9e0d0d',
    red90: '#840f0f',

    yellow10: '#fef7e2',
    yellow60: '#fbc622',
    yellow80: '#fab005',
    yellow90: '#f08c00',

    teal10: '#d6e6e7',
    teal60: '#297e84',
    teal80: '#31696d',
    teal90: '#305558'
  },
  spaces: {
    xxxsmall: 7,
    xxsmall: 10,
    xsmall: 13,
    small: 16,
    medium: 25,
    large: 35,
    xlarge: 45,

    /**
     * After discussing with Ameen,
     * We haven't defined this value globally.
     * This creates inconsistent paddings between pages,
     * or even with elements in a page.
     *
     * spaces.small is the most popular one, then it is choosen.
     */
    gutter: 16
  },
  fonts: {
    fontSans: '"Open Sans", Tahoma, Verdana, Arial, sans-serif',
    fontSerif: 'Georgia, Times, "Times New Roman", serif',
    fontCondensed: '"Open Sans Condensed", Tahoma, Verdana, Arial, sans-serif'
  },
  fontSizes: {
    base: '14px',
    XS: '10px',
    S: '12px',
    M: '14px',
    L: '16px',
    XL: '18px',
    XXL: '22px',
    XXXL: '24px'
  },
  productImageRatio: '143%',
  defaultLineHeight: 1.57
}

const themeExtra = {
  ...theme,
  extra: {
    boxShadow: `0 1px 2px -1px ${theme.colors.colorDark}`
  }
}

export default themeExtra
