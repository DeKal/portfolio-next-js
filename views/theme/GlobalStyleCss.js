import React from 'react'
import PropTypes from 'prop-types'

const GlobalStyleCss = ({ theme }) => {
  return (
    <style jsx global>{`
      html,
      body {
        color: ${theme.colors.colorDark};
      }
      h1 {
        color: ${theme.colors.colorLight};
      }
      a {
        color: ${theme.colors.colorDark};
        border-bottom: 1px dashed ${theme.colors.colorDark};
      }
      body:after {
        content: '';
        position: fixed;
        top: 0;
        bottom: 0;
        width: 100%;
        height: 100%;
        opacity: ${theme.images.bgPatternOpacity};
        z-index: -1;
        background: ${theme.colors.bgColor} url(${theme.images.bgPattern});
      }
      *:selection {
        color: ${theme.colors.colorLight};
        background-color: ${theme.colors.colorDarken};
      }
      *::-moz-selection {
        color: ${theme.colors.colorLight};
        background-color: ${theme.colors.colorDarken};
      }
    `}</style>
  )
}

GlobalStyleCss.propTypes = {
  theme: PropTypes.any
}

export default GlobalStyleCss
