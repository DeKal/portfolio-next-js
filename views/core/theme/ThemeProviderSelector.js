import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components/macro'
import {
  isChristmasDay,
  isNewYearComing,
  isTet,
  isBlogs
} from '~/utils/helpers'
import theme from '~/theme/theme'
import christMasTheme from '~/theme/christMasTheme'
import newYearTheme from '~/theme/newYearTheme'
import tetTheme from '~/theme/tetTheme'
import blogTheme from '~/theme/blogTheme'
import GlobalStyleCss from '~/views/core/theme/GlobalStyleCss'
import { useRouter } from 'next/router'

const ThemeProviderSelector = ({ children }) => {
  const router = useRouter()
  const theme = useTheme(router)
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyleCss theme={theme} />
    </ThemeProvider>
  )
}

ThemeProviderSelector.propTypes = {
  children: PropTypes.any
}

export default ThemeProviderSelector

const useTheme = router => {
  if (isBlogs(router.pathname)) {
    return blogTheme
  }

  if (isChristmasDay()) {
    return christMasTheme
  }

  if (isNewYearComing()) {
    return newYearTheme
  }

  if (isTet()) {
    return tetTheme
  }

  return theme
}
