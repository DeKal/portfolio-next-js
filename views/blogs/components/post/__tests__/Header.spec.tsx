import * as React from 'react'
import { shallowWithBaseTheme } from '~/views/core/utils/withThemeProviders'
import Header from '~/views/blogs/components/post/Header'
import theme from '~/views/core/theme/styleguide/blogTheme'

describe('Blog Header', () => {
  it('should render without throwing an error', function () {
    const wrap = shallowWithBaseTheme(<Header />, theme)
    expect(wrap).toMatchSnapshot()
  })
})
