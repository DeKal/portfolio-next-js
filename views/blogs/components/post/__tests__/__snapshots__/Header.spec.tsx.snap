// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Header should render without throwing an error 1`] = `
<h1
  className="Header-sc-1einucj-0 drTfX"
/>
`;
